### ISSUE TITLE
detect descending runs

### ISSUE THREAD

**Issue by roarin-roran** [roarin-roran/Sorting#77]
currently only ascending runs are detected.

try out adding descending run detection by forking at the start of a run: if the next element is smaller, the run is descending and vice versa.

if this is noticeably slower, consider a new modular option here @sebawild? to measure the effects of forking like this

**Comment by sebawild** [comment]
Yeah; not only can the extra check be slightly slower, we might also want to compare the two variants (also for mergecost).

What does a “modular option” mean in actual Python?

**Comment by roarin-roran** [comment]
currently, we have sorters, mergers, and ipqs - all intended to be mutually intercompatible. I'm considering adding run detectors to that list
